File: svws-webclient/core/src/core/utils/religion/ReligionUtils.ts
import { JavaObject } from '../../../java/lang/JavaObject';
import { IllegalStateException } from '../../../java/lang/IllegalStateException';
import { ReligionEintrag } from '../../../core/data/schule/ReligionEintrag';
import { JavaLong } from '../../../java/lang/JavaLong';
import { JavaString } from '../../../java/lang/JavaString';
import type { Comparator } from '../../../java/util/Comparator';

export class ReligionUtils extends JavaObject {

	/**
	 * Ein Default-Comparator für den Vergleich von Religion-Einträgen.
	 */
	public static readonly comparator : Comparator<ReligionEintrag> = { compare : (a: ReligionEintrag, b: ReligionEintrag) => {
		if ((a.kuerzel === null) || (b.kuerzel === null)) {
			if ((a.kuerzel === null) && (b.kuerzel === null))
				return 0;
			return (a.kuerzel === null) ? -1 : 1;
		}
		const cmp : number = JavaString.compareTo(a.kuerzel, b.kuerzel);
		return (cmp === 0) ? JavaLong.compare(a.id, b.id) : cmp;
	} };

	/**
	 * Ein Comparator für den Vergleich von Religion-Einträgen anhand ihres Textes.
	 */
	public static readonly comparatorText : Comparator<ReligionEintrag> = { compare : (a: ReligionEintrag, b: ReligionEintrag) => {
		if ((a.text === null) || (b.text === null)) {
			if ((a.text === null) && (b.text === null))
				return 0;
			return (a.text === null) ? -1 : 1;
		}
		const cmp : number = JavaString.compareTo(a.text, b.text);
		return (cmp === 0) ? JavaLong.compare(a.id, b.id) : cmp;
	} };


	private constructor() {
		super();
		throw new IllegalStateException("Instantiation not allowed")
	}

	transpilerCanonicalName(): string {
		return 'de.svws_nrw.core.utils.religion.ReligionUtils';
	}

	isTranspiledInstanceOf(name : string): boolean {
		return ['de.svws_nrw.core.utils.religion.ReligionUtils'].includes(name);
	}

}

export function cast_de_svws_nrw_core_utils_religion_ReligionUtils(obj : unknown) : ReligionUtils {
	return obj as ReligionUtils;
}

File: svws-core/src/main/java/de/svws_nrw/core/utils/religion/ReligionUtils.java
package de.svws_nrw.core.utils.religion;

import java.util.Comparator;

import de.svws_nrw.core.data.schule.ReligionEintrag;
import jakarta.validation.constraints.NotNull;

/**
 * Diese Klasse stellt Hilfsmethoden in Bezug auf Religionen zur Verfügung.
 */
public final class ReligionUtils {

	private ReligionUtils() {
		throw new IllegalStateException("Instantiation not allowed");
	}


	/** Ein Default-Comparator für den Vergleich von Religion-Einträgen. */
	public static final @NotNull Comparator<ReligionEintrag> comparator =
			(final @NotNull ReligionEintrag a, final @NotNull ReligionEintrag b) -> {
				int cmp = a.sortierung - b.sortierung;
				if (cmp != 0)
					return cmp;
				if ((a.kuerzel == null) || (b.kuerzel == null)) {
					if ((a.kuerzel == null) && (b.kuerzel == null))
						return 0;
					return (a.kuerzel == null) ? -1 : 1;
				}
				cmp = a.kuerzel.compareTo(b.kuerzel);
				return (cmp == 0) ? Long.compare(a.id, b.id) : cmp;
			};

	/** Ein Comparator für den Vergleich von Religion-Einträgen anhand ihres Textes. */
	public static final @NotNull Comparator<ReligionEintrag> comparatorText =
			(final @NotNull ReligionEintrag a, final @NotNull ReligionEintrag b) -> {
				if ((a.text == null) || (b.text == null)) {
					if ((a.text == null) && (b.text == null))
						return 0;
					return (a.text == null) ? -1 : 1;
				}
				final int cmp = a.text.compareTo(b.text);
				return (cmp == 0) ? Long.compare(a.id, b.id) : cmp;
			};

}
